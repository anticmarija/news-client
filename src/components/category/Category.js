import React, { useEffect } from "react";
import { useParams } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import { fetchAllArticlesPerCategory } from "../../store/actions/categories.actions";
import Thumbnail from "../thumbnail/Thumbnail";
import Loader from "../loader/Loader";
import { MainHeaderStyle, NewsWrapperStyle } from "../../style/Shared.style";
import { capitalize } from "../../utils/helpers";

const Category = () => {
  let { activeCountry } = useSelector((state) => state.countries);

  let { categoryName } = useParams();
  let { loadingAllCategoryArticles, allCategoryArticles } = useSelector(
    (state) => state.categories
  );

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchAllArticlesPerCategory(activeCountry, categoryName));
    // eslint-disable-next-line
  }, [activeCountry, categoryName]);

  return (
    <div>
      <MainHeaderStyle>Category {capitalize(categoryName)}</MainHeaderStyle>
      {loadingAllCategoryArticles ? (
        <Loader />
      ) : (
        <NewsWrapperStyle>
          {allCategoryArticles.map((article) => (
            <Thumbnail
              key={`${article.title}_${article.publishedAt}`}
              article={article}
            />
          ))}
        </NewsWrapperStyle>
      )}
    </div>
  );
};

export default Category;
