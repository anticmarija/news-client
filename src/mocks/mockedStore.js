import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const middlewares = [thunk];
export const initialState = {
  topNews: {},
  countries: {
    supportedCountries: [{ countryId: "gb", countryName: "Gb" }],
    activeCountry: "gb",
    isChangingCountriesEnabled: true,
  },
  categories: {
    allCategoryArticles: [
      { title: "article1", description: "some desc" },
      { title: "article2", description: "some desc" },
    ],
    categoryArticles: {
      business: [
        { title: "article1", description: "some desc" },
        { title: "article2", description: "some desc" },
      ],
      sports: [
        { title: "article3", description: "some desc" },
        { title: "article4", description: "some desc" },
      ],
    },
  },
  article: {
    activeArticle: {
      title: "active",
      urlToImage:
        "https://i.dailymail.co.uk/1s/2020/06/17/16/29719992-0-image-a-23_1592408042536.jpg",
      description: "desc",
    },
  },
};

export const mockedStore = (state = initialState) => {
  const mockStore = configureStore(middlewares);
  return mockStore(state);
};
