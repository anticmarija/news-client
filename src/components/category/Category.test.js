import React from "react";
import Category from "./Category";
import { mockedStore } from "../../mocks/mockedStore";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

describe("Test Category component", () => {
  it("should render correctly", () => {
    const store = mockedStore();

    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Category />
        </MemoryRouter>
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
