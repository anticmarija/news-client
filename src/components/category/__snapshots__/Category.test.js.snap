// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Category component should render correctly 1`] = `
<div>
  <h1
    class="Sharedstyle__MainHeaderStyle-sc-1a3bnp9-0 chhIjm"
  >
    Category 
    
  </h1>
  <div
    class="Sharedstyle__NewsWrapperStyle-sc-1a3bnp9-1 jTferJ"
  >
    <div
      class="Thumbnailstyle__ThumbnailWrapperStyle-sc-123obdz-0 lhHvSI"
    >
      <h2
        class="Thumbnailstyle__ThumbnailTitleStyle-sc-123obdz-3 dSnuxn"
      >
        article1
      </h2>
      <div
        class="Thumbnailstyle__ThumbnailImgStyle-sc-123obdz-1 cwXTzT"
      />
      <div
        class="Thumbnailstyle__ThumbnailDescStyle-sc-123obdz-2 CqRvk"
      >
        <div>
          some desc
        </div>
        <span
          class="Thumbnailstyle__ThumbnailLinkStyle-sc-123obdz-4 VvDHO"
        >
          <a
            data-testid="more-link"
            href="/country/gb/articles/article1"
          >
            More &gt;
          </a>
        </span>
      </div>
    </div>
    <div
      class="Thumbnailstyle__ThumbnailWrapperStyle-sc-123obdz-0 lhHvSI"
    >
      <h2
        class="Thumbnailstyle__ThumbnailTitleStyle-sc-123obdz-3 dSnuxn"
      >
        article2
      </h2>
      <div
        class="Thumbnailstyle__ThumbnailImgStyle-sc-123obdz-1 cwXTzT"
      />
      <div
        class="Thumbnailstyle__ThumbnailDescStyle-sc-123obdz-2 CqRvk"
      >
        <div>
          some desc
        </div>
        <span
          class="Thumbnailstyle__ThumbnailLinkStyle-sc-123obdz-4 VvDHO"
        >
          <a
            data-testid="more-link"
            href="/country/gb/articles/article2"
          >
            More &gt;
          </a>
        </span>
      </div>
    </div>
  </div>
</div>
`;
